src/ir/utils.rs

```diff
@@ -578,6 +582,36 @@ where
     }
 }
 
+impl<T, C> IdxLike<T> for Foreign<T, C>
+where
+    C: Ctx<T>,
+{
+    const UNKNOWN: Self = Foreign {
+        key: Idx::UNKNOWN,
+        owner: Idx::UNKNOWN,
+    };
+
+    fn new(idx: usize) -> Self {
+        Self {
+            key: Idx::new(idx),
+            owner: Idx::UNKNOWN,
+        }
+    }
+
+    fn get(self) -> usize {
+        self.key.get()
```

Comment:
This should be discussed as it seems somewhat dangerous, but I can't think of a better method that is useful.

Reply:
`IdxLike` requires you to implement this method so we don't really have a choice but I see your point. Ideally we'd want to scope the methods of the trait so that the only way to use them is within the containers but trait methods are visible everywhere AFAIK.